### PR TITLE
Path filter in dynamic profile panel

### PR DESCRIPTION
**Pull request by nicolas-t**
Hello,

I started working on adding path filter to the dynamic profile panel.
I don't think I have enough knowledge to do more, so if you like this feature please feel free to finish the job.

Related issue: https://github.com/Beep6581/RawTherapee/issues/6281

### CODE REVIEW

**Comment by Thanatomanic** [comment]
Thanks @nicolas-t for working on your own idea! Let us know when there's something to be tested or you need advice.

**Comment by nicolas-t** [comment]
Hello @Thanatomanic  :)

I'm having trouble accessing the filename in the `matches` function here : 
https://github.com/Beep6581/RawTherapee/blob/1e797f11e288d0f3dd6bd4e1e4464d9f869ab430/rtengine/dynamicprofile.cc#L89

--- 

In this function I think I have `inputFile` which is the path i'm looking for: 
https://github.com/Beep6581/RawTherapee/blob/b35e0d82694c3d5a80c5decdcd2ddb096f505a07/rtgui/main-cli.cc#L744

But, should I pass it as the second argument of `loadDynamicProfile`, and then pass it again as the second argument of [`rule.matches`](https://github.com/Beep6581/RawTherapee/blob/b35e0d82694c3d5a80c5decdcd2ddb096f505a07/rtengine/profilestore.cc#L524) ?

Seems better to me to store the filename in the `FramesMetaData`/ `FramesData`, but I'm not sure how.


Also, I never did any c++, how can I dump/echo the value of a variable ? 
 
Thanks for you help.

**Comment by Lawrence37** [comment]
Hi.
I don't think the file name goes in the metadata, so passing it as an argument is the better option. You can inspect variable values with `std::cout`/`printf()` or by using a debugger such as GDB on the debug build.

**Comment by nicolas-t** [comment]
Hello @Lawrence37 !
I just pushed 2 commits, I'm passing the filename as the second argument.
App is working normally, and my dynamic profile is well applied depending on filename and path.

Aside from this I tried outputing the filename here : 
https://github.com/Beep6581/RawTherapee/pull/6284/files#diff-bc5e968d23075d5dae37c689a2080f5441c91eb1d3618ab661af1be30cad5c7dR740

But nothing is showing up the terminal.
I'm compiling like this : 
`~/programs/code-rawtherapee/tools/build-rawtherapee -b -d`

And running like this : 
`~/programs/rawtherapee-dev-debug/rawtherapee`
Out of curiosity: where is the output supposed to appear ? 

Anyway, the feature is working on my side, can you test and review my code ?

Thanks

**Comment by Lawrence37** [comment]
The output appears in the terminal. You've placed the code in the CLI version of RT, so it will only appear if you run `rawtherapee-cli` instead of `rawtherapee`.

I briefly tested this feature and it works fine. I will add my code review comments shortly.

File: rtdata/languages/English (US) [hunk 1]
```diff
@@ -62,6 +62,7 @@
 !EXIFFILTER_ISO;ISO
 !EXIFFILTER_LENS;Lens
 !EXIFFILTER_METADATAFILTER;Enable metadata filters
+!EXIFFILTER_PATH;Path
```
Comment (Lawrence37) [inline rtdata/languages/English (US)]: This is usually added by running a script, but I think it's fine? Ping @Beep6581.

File: rtdata/languages/default [hunk 2]
```diff
@@ -61,6 +61,7 @@ EXIFFILTER_IMAGETYPE;Image type
 EXIFFILTER_ISO;ISO
 EXIFFILTER_LENS;Lens
 EXIFFILTER_METADATAFILTER;Enable metadata filters
+EXIFFILTER_PATH;Path
 EXIFFILTER_SHUTTER;Shutter
```
Comment (Lawrence37) [inline rtdata/languages/default:65]: I'm concerned that "Path" is a little vague. Something like "File path" is more clear.

File: rtgui/main-cli.cc [hunk 3]
```diff
@@ -737,11 +737,13 @@ int processLineParams ( int argc, char **argv )
         }
 
         if (useDefault) {
+            std::cout << "loading file: " << inputFile << std::endl;
```
Comment (Lawrence37) [inline rtgui/main-cli.cc]: The file isn't being loaded here.

**Review by Lawrence37 — DISMISSED** [review]

**Review by Lawrence37 — COMMENTED** [review]
@nicolas-t This looks good.

Should this be included in 5.9? It's a small change but the beta version is almost here and the default language file is under review.

**Comment by Thanatomanic** [comment]
@nicolas-t Thanks for the work and the patience!

**Pull request merged**